agent/agent.go

```diff
@@ -59,17 +59,20 @@ func _main() int {
 
 	log.Info("Loading configuration")
 	cfg, err := config.NewConfig()
-	if err != nil {
-		log.Errorf("Error loading config: %v", err)
-		return exitcodes.ExitTerminal
-	}
-
+	// Load cfg before doing 'versionFlag' so that it has the DOCKER_HOST
+	// variable loaded if needed
 	if *versionFlag {
 		versionableEngine := engine.NewTaskEngine(cfg)
 		version.PrintVersion(versionableEngine)
 		return exitcodes.ExitSuccess
 	}
 
+	if err != nil {
+		log.Criticalf("Error loading config: %v", err)
+		return exitcodes.ExitTerminal
```

Comment:
Do we need a distinction between retry-able config errors (can't read from metadata service because it's not up yet) versus total failures (cannot parse)?  Your only fatal one right now is AWSRegion which can potentially be retry-able.